### ISSUE TITLE
Sequential run of terraform init lead to "Backend configuration changed" error

### ISSUE THREAD

**Issue by denis256** [hashicorp/terraform#34066]
### Terraform Version

```shell
Terraform v1.6.1
on linux_amd64
```

### Terraform Configuration Files

```terraform
terraform {
  backend "s3" {}
}
```

### Debug Output

https://gist.github.com/denis256/79d71388780e70f16d4bd7bacee9c276

### Expected Behavior

No errors to be generated

### Actual Behavior

On Terraform 1.6.1, occurs error
```
╷
│ Error: Backend configuration changed
│ 
│ A change in the backend configuration has been detected, which may require migrating existing state.
│ 
│ If you wish to attempt automatic migration of the state, use "terraform init -migrate-state".
│ If you wish to store the current configuration with no changes to the state, use "terraform init -reconfigure".
╵
```
On Terraform 1.5.7 no errors occur

### Steps to Reproduce

1. terraform init -backend-config=bucket=existing-s3-bucket -backend-config=encrypt=true -backend-config=key=qwe.tfstate -backend-config=region=us-west-2
1. terraform init -backend-config=bucket=existing-s3-bucket -backend-config=encrypt=true -backend-config=key=qwe.tfstate -backend-config=region=us-west-2

### Additional Context

Example in https://github.com/denis256/terragrunt-tests/tree/master/tf-init-error

**Comment by noahnu** [comment]
Possibly related: https://github.com/hashicorp/terraform/issues/30495 (I'm encountering this as well on terraform 1.6.0)

**Comment by jbardin** [comment]
Closed via upstream bug fix in #34108

**Comment by lorengordon** [comment]
@jbardin What version of Terraform will include that fix?

**Comment by jbardin** [comment]
@lorengordon, that PR was included in v1.6.2  https://github.com/hashicorp/terraform/releases/tag/v1.6.2

**Comment by lorengordon** [comment]
Thank you! I tried to follow the thread of commits, but it didn't seem very clear for some reason.
